app/utils/CurrentDateTime.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import javax.inject.{Inject, Singleton}
+import org.joda.time.{DateTime, DateTimeZone}
+
+@Singleton
+class CurrentDateTime @Inject()() {
```

Comment:
The name suggests this is a `DateTime`. It provides a `DateTime` but it isn't one itself. It's really a clock.
```suggestion
class Clock @Inject()() {
```
(As a possible alternative, note there is already a standard `Clock` class in the `java.time` package which also allows instances where the time can be given a specific value e.g. for use within unit tests.)